examples/imagenet/imagenet_lib.py

```diff
@@ -191,58 +159,78 @@ def _prepare(x):
   return jax.tree_map(_prepare, xs)
 
 
-def create_input_iter(batch_size, image_size, dtype, train, cache):
-  ds = input_pipeline.load_split(
-      batch_size, image_size=image_size, dtype=dtype, train=train, cache=cache)
-  it = map(prepare_tf_data, ds)
+def _create_input_iter(dataset_builder, batch_size, image_size, dtype, train,
+                       cache):
+  ds = input_pipeline.create_split(
```

Comment:
Can you tell me more about this TFDS API change? What are "dataset builders"?

Reply:
The reason for this was majorly because to get rid of previously globally declared vars:
TRAIN_IMAGES = 1281167
EVAL_IMAGES = 50000

To have an extendable solution, we shouldn't have hardcoded values of something which is controlled by a different library. In order to fetch this information using the TDFS library itself, I modified the existing code to use DatasetBuilder (an abstract base class for every dataset). DatasetBuilder contains information around dataset such as: feature names, types, shapes, version, splits etc.

Also the dataset_builder is initialized in the main train loop to avoid multiple initializations which increases the chances of inconsistent changes.